crypto/CMakeLists.txt

```diff
@@ -101,6 +101,10 @@ if(PERL_EXECUTABLE)
   perlasm(cipher_extra/aes128gcmsiv-x86_64.${ASM_EXT} cipher_extra/asm/aes128gcmsiv-x86_64.pl)
   perlasm(cipher_extra/chacha20_poly1305_x86_64.${ASM_EXT} cipher_extra/asm/chacha20_poly1305_x86_64.pl)
   perlasm(cipher_extra/chacha20_poly1305_armv8.${ASM_EXT} cipher_extra/asm/chacha20_poly1305_armv8.pl)
+  if(NOT MY_ASSEMBLER_IS_TOO_OLD_FOR_AVX)
```

Comment:
Yeah this is also one way to do to. But I think we should stick to the current method we use, to make it uniform.

Reply:
Discussed with Torben, the approach can only partially disable the assembly code guarded by `avx` but the use case is to disable all assembly code.

Reply:
Discussed with Dusan, `MY_ASSEMBLER_IS_TOO_OLD_FOR_AVX` is removed so the snitch code will get built without `avx` because `sse` is also available.

Reply:
Looks like I still need to disable the whole file due to below error caught by gcc 4.1 build.

```
/codebuild/output/src612645169/src/github.com/awslabs/aws-lc/test_build_dir/crypto/cipher_extra/aesni-sha256-x86_64.S:118: Error: no such instruction: `aesenc %xmm4,%xmm6'
--
408 | /codebuild/output/src612645169/src/github.com/awslabs/aws-lc/test_build_dir/crypto/cipher_extra/aesni-sha256-x86_64.S:122: Error: no such instruction: `aesenc %xmm5,%xmm6'

```

Reply:
CryptoAlg-1091 is created to track `|EVP_aes_128/256_cbc_hmac_sha1/256|` API performance on gcc 4.1
```
# CryptoAlg-1091:
    # Stitch code |aesni-sha1/256-x86_64.${ASM_EXT}| are not enabled due
    # to some build issues and lack of tools to measure the performance gap.
    # The disable is safer choice because |EVP_aes_128/256_cbc_hmac_sha1/256|
    # are deprecated.
```